todo_app/tests/test_requests.py

```diff
@@ -1,49 +1,98 @@
+        Item.item_description: "My cool task 3"
+    }
 
-def mock_responses(method, url, **kwargs):
-    response = Mock()
-    if url.startswith('https://api.trello.com/1/boards/%s/cards' % mockresponses.board_id):
-        response.json.return_value = mockresponses.cards
-        return response
-        
-    elif url.startswith('https://api.trello.com/1/members/%s/boards' % os.getenv('TRELLO_USER')):
-        response.json.return_value = mockresponses.boards
-        return response
+task_to_be_deleted = {
+        Item.item_title: "Task 4",
+        Item.item_due_date: "22/03/2021",
+        Item.item_description: "My cool task 4",
+        Item.item_status: BoardStatus._TODO
+    }
 
+task_to_be_moved = {
+        Item.item_title: "Task 5",
+        Item.item_due_date: "29/03/2021",
+        Item.item_description: "My cool task 5",
+        Item.item_status: BoardStatus._TODO
+    }
 
-    return None
-
+def get_db_collection():
+    db_conn = os.getenv('DB_CONNECTION_STRING')
+    client = pymongo.MongoClient(db_conn)
+    return client.todoapp['tasks']
 
 @pytest.fixture
-def client():
+def app_client():
     # Use our test integration config instead of the 'real' version
     file_path = find_dotenv('.env.test')
     load_dotenv(file_path, override=True)
 
-    # Create the new app.
-    test_app = app.create_app()
-    # Use the app to create a test_client that can be used in our tests.
-    with test_app.test_client() as client:
-        yield client  
+    # Initialise mongomock here.
+    with mongomock.patch(servers=(('fakemongo', 27017),)):
+        # Create the new app.
+        test_app = app.create_app()
+        # Use the app to create a test_client that can be used in our tests.
+        with test_app.test_client() as client:
+            yield client
 
 
-@patch('requests.request')
-def test_index_page(mock_get_requests, client):
-    mock_get_requests.side_effect = mock_responses
+def test_show_tasks(app_client):
+    tasks = get_db_collection() 
+    tasks.insert_many(initial_mock_tasks)
+    
+    response = app_client.get('/')
+    assert response.status_code == 200 
+    assert tasks.find({Item.item_title: "Task 1"}) is not None
 
-    response = client.get('/')
+def test_add_task(app_client):
+    tasks = get_db_collection() 
+
+    response = app_client.post('/add', data=dict(new_task))
     assert response.status_code == 200 
+    assert tasks.find({Item.item_title: "Task 3"}) is not None
+
+def test_delete_task(app_client):
+    tasks = get_db_collection() 
+    inserted_id = tasks.insert_one(task_to_be_deleted).inserted_id
     
-    html_string = str(response.data)
-    assert 'Tempting to say' in html_string
-    assert html_string.count('status_0') == 4 # 4 in todo
-    assert html_string.count('status_1') == 3 # 3 doing
-    assert html_string.count('status_2') == 5 # 5 done
\ No newline at end of file
+    assert tasks.find_one({Item.item_id: inserted_id}) is not None
```

Comment:
This is assertion is just testing that mongomock works correctly. You can delete this. Similarly for test_move_task

Reply:
I was thinking of keeping as a proof that the function works as expected